apps/core/app/[locale]/(default)/login/_components/change-password-form.tsx

```diff
@@ -0,0 +1,165 @@
+'use client';
+
+import { Button } from '@bigcommerce/components/button';
+import {
+  Field,
+  FieldControl,
+  FieldLabel,
+  FieldMessage,
+  Form,
+  FormSubmit,
+} from '@bigcommerce/components/form';
+import { Input } from '@bigcommerce/components/input';
+import { Message } from '@bigcommerce/components/message';
+import { Loader2 as Spinner } from 'lucide-react';
+import { useTranslations } from 'next-intl';
+import { ChangeEvent, useRef, useState } from 'react';
+import { useFormStatus } from 'react-dom';
+
+import { useRouter } from '~/navigation';
+
+import { submitChangePasswordForm } from '../_actions/submit-change-password-form';
+
+interface Props {
+  customerId: number;
+  customerToken: string;
+}
+
+interface FormStatus {
+  status: 'success' | 'error';
+  message: string;
+}
+
+const SubmitButton = () => {
+  const { pending } = useFormStatus();
+  const t = useTranslations('Account.SubmitChangePassword');
+
+  return (
+    <Button
+      className="relative w-fit items-center px-8 py-2"
+      data-button
+      disabled={pending}
+      variant="primary"
+    >
+      <>
+        {pending && (
+          <>
+            <span className="absolute z-10 flex h-full w-full items-center justify-center bg-gray-400">
+              <Spinner aria-hidden="true" className="animate-spin" />
+            </span>
+            <span className="sr-only">{t('spinnerText')}</span>
+          </>
+        )}
+        <span aria-hidden={pending}>{t('submitText')}</span>
+      </>
+    </Button>
+  );
+};
+
+export const ChangePasswordForm = ({ customerId, customerToken }: Props) => {
+  const form = useRef<HTMLFormElement>(null);
+  const router = useRouter();
+  const [formStatus, setFormStatus] = useState<FormStatus | null>(null);
+  const [newPassword, setNewPasssword] = useState('');
+  const [isNewPasswordValid, setIsNewPasswordValid] = useState(true);
+  const [isConfirmPasswordValid, setIsConfirmPasswordValid] = useState(true);
+
+  const t = useTranslations('Account.ChangePassword');
+
+  const handleNewPasswordValidation = (e: ChangeEvent<HTMLInputElement>) => {
+    const validationStatus = e.target.validity.patternMismatch;
+
+    setNewPasssword(e.target.value);
+
+    return setIsNewPasswordValid(!validationStatus);
+  };
+
+  const handleConfirmPasswordValidation = (e: ChangeEvent<HTMLInputElement>) => {
+    const confirmPassword = e.target.value;
+
+    return setIsConfirmPasswordValid(confirmPassword === newPassword);
+  };
+
+  const onSubmit = async (formData: FormData) => {
+    const submit = await submitChangePasswordForm({ formData, customerId, customerToken });
+
+    if (submit.status === 'success') {
+      form.current?.reset();
+      setFormStatus({
+        status: 'success',
+        message: t('successMessage'),
+      });
+
+      setTimeout(() => router.push('/login'), 2000);
+    }
+
+    if (submit.status === 'failed') {
+      setFormStatus({ status: 'error', message: submit.error ?? '' });
+    }
```

Comment:
Should we just use the same `status` strings between the action and the component? What if `submitChangePasswordForm` returned `status: 'error'` when an error occurred? It could potentially simplify this 🙂

Reply:
Good point, Matt. I've refactored it already